### PR TITLE
fix(postgresql_info): properly quote args to SHOW to avoid triggering 'reserved word' errors

### LINKED ISSUE #314: postgresql_info show command with protected keywords
##### SUMMARY

Show command in postgresql_info is not protected by brackets, when a parameter contain a key word (like analyze), it raises an SQL syntax error.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Module postgresql_info (it sould be in this line [here](https://github.com/ansible-collections/community.postgresql/blob/7de6946b60cdf1301f77aaaf5add605a354873ba/plugins/modules/postgresql_info.py#L1038))
##### ANSIBLE VERSION

```paste below
  config file = /home/iac/git/deploy-postgres/ansible.cfg
  configured module search path = ['/home/iac/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /home/iac/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
community.postgresql 2.1.5
```

##### CONFIGURATION

```paste below
DEFAULT_LOAD_CALLBACK_PLUGINS(env: ANSIBLE_LOAD_CALLBACK_PLUGINS) = True
DEFAULT_STDOUT_CALLBACK(/home/iac/git/deploy-postgres/ansible.cfg) = debug
HOST_KEY_CHECKING(/home/iac/git/deploy-postgres/ansible.cfg) = False
RETRY_FILES_ENABLED(/home/iac/git/deploy-postgres/ansible.cfg) = False
```

##### OS / ENVIRONMENT

OS Ubuntu 20.04.4 LTS
Server PostgreSQL 13.7 (Ubuntu 13.7-1.pgdg20.04+1)
With the extension postgresql-13-partman/focal-pgdg,now 4.6.2-1.pgdg20.04+1 amd64

##### STEPS TO REPRODUCE

In order to reproduce this bug you need to have `shared_preload_libraries='pg_partman_bgw'` on postgresql.conf file.
This will add `pg_partman_bgw.analyze` parametter to the database (among many others)

And then run postgresql_info module (with postgreSQL running on default port and socket directory)
```yaml
- name: Get detailed informations of PostgreSQL
  community.postgresql.postgresql_info:
  become_user: postgres
  become: yes
```

##### EXPECTED RESULTS

Module succed with all information about the instance of PostgreSQL

##### ACTUAL RESULTS

```paste below
fatal: [target]: FAILED! => {
    "changed": false
}

MSG:

Cannot execute SQL 'SHOW pg_partman_bgw.analyze': syntax error at or near "analyze"
LINE 1: SHOW pg_partman_bgw.analyze
```
##### OTHERS INFOMATIONS
`SHOW pg_partman_bgw.analyze` command raise an error but protected with brackets like that `SHOW "pg_partman_bgw.analyze"` doesn't, because the "analyse" keyword is protected.

I changed `return self.__exec_sql("SHOW %s" % setting)[0][0]` into `return self.__exec_sql('SHOW "%s"' % setting)[0][0]` in [here ](https://github.com/ansible-collections/community.postgresql/blob/4d24c4cfa8f37efafc243d53cf6868967952fecd/plugins/modules/postgresql_info.py#L1035) and it doesn't raise any error upon execution, i think it could resolve the issue

### PR DESCRIPTION
**Pull request by hunleyd**
Co-authored-by: @Jhiliano

##### SUMMARY
Modify postgresql_info to properly quote arguments to the `SHOW` command so that PG doesn't run it through the parser and complain about reserved keywords.

Fixes #314 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgresql_info

### CODE REVIEW

**Comment by hunleyd** [comment]
hey @Jhiliano / @keithf4 can you see if this meets your needs? thx!

File: changelogs/fragments/314-postgresql_info-quote-show-args.yml
Comment (keithf4) [inline changelogs/fragments/314-postgresql_info-quote-show-args.yml:2]: Wasn't just issue with reserved words. I believe we ran into this because of special characters since almost all third-party extensions have periods in their variable names to avoid collisions.

**Review by keithf4 — CHANGES_REQUESTED** [review]

**Comment by Jhiliano** [comment]
It's good for me, i tested this exact fix and i didn't run into any others issues

**Comment by hunleyd** [comment]
thanks @Jhiliano! i'll merge and backport as soon as i can get an approval from the team. thanks again for reporting the issue and the fix.

**Review by jchancojr — APPROVED** [review]
lgtm

**Comment by Andersson007** [comment]
oh, just noticed, thank you all for your contributions!

**Pull request merged**
